src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/MergeReservation.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Commands.Reservations.Common;
+using Microsoft.Azure.Management.Reservations.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Reservations;
+using Microsoft.Azure.Commands.Reservations.Models;
+
+namespace Microsoft.Azure.Commands.Reservations.Cmdlets
+{
+    [Cmdlet(VerbsData.Merge, "AzureRmReservation", DefaultParameterSetName = Constants.ParameterSetNames.CommandParameterSet, SupportsShouldProcess = true), OutputType(typeof(List<PSReservation>))]
+    public class MergeReservation : AzureReservationsCmdletBase
+    {
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet, 
+            Mandatory = true)]
+        [ValidateNotNull]
+        public string ReservationOrderId { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet, 
+            Mandatory = true)]
+        [ValidateNotNull]
+        [ValidateCount (2, 2)]
+        public string[] ReservationId { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.ObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true)]
+        [ValidateNotNull]
+        [ValidateCount(2, 2)]
+        public PSReservation[] Reservation { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess("AzureRmReservation", "Merge"))
```

Comment:
Is this allyou want the user to know about the actions of this cmdlet?  What is the impact of merging two reservations?

Reply:
this was originally not in our code but I had to write it to pass static analysis. (I think it asked me to use ShouldProcess(string, string)) There were several overloaded functions, we may consider using those ones in next sprint fix.

Reply:
ShouldProcess is required.  You need to provide meaningful strings to SHouldProcess.  The users use this to determine what would happen if they run a script without actually making changes.  The minimum amount of information is to provide the action and identity of the object (resource id or parent name + name are acceptable for this)